### PR TITLE
doc: update project documentation

### PR DESCRIPTION
**Pull request by dvrogozh**
This patch improves top level documentation focusing on easy examples and moving more difficult stuff to separate documents.

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -3,515 +3,314 @@ Media Delivery Software Stack
...
++----------------------------------------------------+-----------------------------+------------------------------------------------+--------------------------------------------+
 
-Once host is setup to access GPU, make sure to install docker engine::
+.. _docker/ubuntu20.04/selfbuild/Dockerfile: docker/ubuntu20.04/selfbuild/Dockerfile
+.. _docker/ubuntu20.04/selfbuild-prodkmd/Dockerfile: docker/ubuntu20.04/selfbuild-prodkmd/Dockerfile
+.. _docker/ubuntu20.04/native/Dockerfile: docker/ubuntu20.04/native/Dockerfile
 
-  # If you run Ubuntu 20.04 on your host:
-  apt-get install docker.io
+To build any of configurations, first clone Media Delivery repository::
```
Comment (dsocek) [inline README.rst:195]: the configurations

**Review by dsocek — APPROVED** [review]
Looks good.  Suggested few minor changes before merge

File: doc/demo.rst [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    -p 8080:8080 \
+    intel-media-delivery demo -4 \
+      http://localhost:8080/vod/avc/WAR_TRAILER_HiQ_10_withAudio-1/index.m3u8
+      http://localhost:8080/vod/avc/WAR_TRAILER_HiQ_10_withAudio-2/index.m3u8
+      http://localhost:8080/vod/avc/WAR_TRAILER_HiQ_10_withAudio-3/index.m3u8
+      http://localhost:8080/vod/avc/WAR_TRAILER_HiQ_10_withAudio-4/index.m3u8
+
+"CDN" sample can be further scaled. For example, transcoding requests might be served
+by the dedicated system where server similar to socat one is running.
+Furthermore, each transcoding might be done on the dedicated GPU-capable system
```
[the remark this answers is not in the document]
Comment (dvrogozh) [reply doc/demo.rst]: done

File: README.rst (hunk 1 above)
Comment (dvrogozh) [reply README.rst]: done

**Pull request merged**